rmw_fastrtps_cpp/src/rmw_logging.cpp

```diff
@@ -0,0 +1,56 @@
+// Copyright 2016 Proyectos y Sistemas de Mantenimiento SL (eProsima).
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "rmw/rmw.h"
+
+#include "fastrtps/log/Log.h"
+
+extern "C"
+{
+using eprosima::fastrtps::Log;
+
+eprosima::fastrtps::Log::Kind convert_rmw_severity_type(rmw_log_severity_t severity)
+{
+  switch(severity)
+  {
+  case RMW_LOG_SEVERITY_WARN:
+       return Log::Kind::Warning;
+  case RMW_LOG_SEVERITY_INFO:
+       return Log::Kind::Info;
+/* Fast-RTPS supports the following logging 'Kind's.
+ * Error : Max priority
+ * Warning : Medium priority
+ * Info : Low priority
+ * From rmw logging severity there are FATAL & DEBUG types as well
+ * We map them to ERROR type of Fast-RTPS which has maximum priority
+ */
+  case RMW_LOG_SEVERITY_ERROR:
+  case RMW_LOG_SEVERITY_FATAL:
+  case RMW_LOG_SEVERITY_DEBUG:
+       return Log::Kind::Error;
+  default:
+       // Fallback to Info if undefined types
+       return Log::Kind::Info;
+  }
+}
+
+rmw_ret_t
+rmw_set_log_severity(rmw_log_severity_t severity)
+{
+  Log::Kind _severity = convert_rmw_severity_type(severity);
+  eprosima::fastrtps::Log::SetVerbosity(_severity);
```

Comment:
If the severity can't be mapped (the `default` case in the other function) it might be better to print an error message rather than setting a fallback level without letting the user know.

Reply:
@dirk-thomas  ok. In that case I'll have to change the API to return a type rmw_ret_t. And, may I use RMW_SET_ERROR_MSG for error setting on unknown types? There has been instances where inside rmw RCUTILS macro for error logging has been used. Let me know on this.

Reply:
@dirk-thomas Could you let me know your thoughts on the previous comment?

Reply:
Since this function is private, only being used once in the same .cpp file, and the calling function is only a couple of lines ong I would suggest to just remove the separate function and move the implementation into `rmw_set_log_severity`. That would simplify the code. `rmw_set_log_severity` can then indeed return an error code and set the error message accordingly.

Reply:
Ok. Thanks. I'll make the necessary changes and update the commit.